data-service/src/test/java/com/amazonaws/blox/dataservice/api/DescribeEnvironmentApiTest.java

```diff
@@ -0,0 +1,142 @@
+import org.junit.runner.RunWith;
+import org.mapstruct.factory.Mappers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.time.Instant;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.StrictStubs.class)
+public class DescribeEnvironmentApiTest {
+  private static final String ACCOUNT_ID = "123456789012";
+  private static final String CLUSTER = "cluster";
+  private static final String ENVIRONMENT_NAME = "name";
+  private static final String ROLE_ARN = "role";
+  private static final String ACTIVE_ENVIRONMENT_REVISION_ID = "123456789012_cluster_name";
+  private static final String DEPLOYMENT_METHOD = "deploymentMethod";
+
+  @Mock private EnvironmentRepository environmentRepository;
+  @Captor private ArgumentCaptor<EnvironmentId> environmentIdArgumentCaptor;
+
+  private Environment environment;
+  private EnvironmentId environmentId;
+  private com.amazonaws.blox.dataservicemodel.v1.model.EnvironmentId environmentIdWrapper;
+  private DescribeEnvironmentRequest describeEnvironmentRequest;
+  private DescribeEnvironmentApi describeEnvironmentApi;
+  private ApiModelMapper apiModelMapper = Mappers.getMapper(ApiModelMapper.class);
+
+  @Before
+  public void setup() {
+    describeEnvironmentApi = new DescribeEnvironmentApi(apiModelMapper, environmentRepository);
+    environmentId =
+        EnvironmentId.builder()
+            .accountId(ACCOUNT_ID)
+            .cluster(CLUSTER)
+            .environmentName(ENVIRONMENT_NAME)
+            .build();
+    environmentIdWrapper =
+        com.amazonaws.blox.dataservicemodel.v1.model.EnvironmentId.builder()
+            .accountId(ACCOUNT_ID)
+            .cluster(CLUSTER)
+            .environmentName(ENVIRONMENT_NAME)
+            .build();
+    environment =
+        Environment.builder()
+            .environmentId(environmentId)
+            .role(ROLE_ARN)
+            .environmentType(EnvironmentType.Daemon)
+            .environmentStatus(EnvironmentStatus.Inactive)
+            .environmentHealth(EnvironmentHealth.Healthy)
+            .createdTime(Instant.now())
+            .lastUpdatedTime(Instant.now())
+            .activeEnvironmentRevisionId(ACTIVE_ENVIRONMENT_REVISION_ID)
+            .latestEnvironmentRevisionId(ACTIVE_ENVIRONMENT_REVISION_ID)
+            .deploymentConfiguration(new DeploymentConfiguration())
+            .deploymentMethod(DEPLOYMENT_METHOD)
+            .validEnvironment(true)
+            .build();
+    describeEnvironmentRequest =
+        DescribeEnvironmentRequest.builder().environmentId(environmentIdWrapper).build();
+  }
+
+  @Test
+  public void describeEnvironmentSuccess() throws Exception {
+    when(environmentRepository.getEnvironment(isA(EnvironmentId.class))).thenReturn(environment);
```

Comment:
could minimal the condition to `when(environmentRepository.getEnvironment(environmentId))`

Reply:
Fixed!